mpi3501.cpp

```diff
@@ -1,135 +1,152 @@
+  SET_GPIO_MODE(GPIO_TFT_RESET_PIN, 1);
+  SET_GPIO(GPIO_TFT_RESET_PIN);
+  usleep(120 * 1000);
+  CLEAR_GPIO(GPIO_TFT_RESET_PIN);
+  usleep(120 * 1000);
+  SET_GPIO(GPIO_TFT_RESET_PIN);
+  usleep(120 * 1000);
+#endif
+
+  // For sanity, start with both Chip selects high to ensure that the display will see a high->low enable transition when we start.
+  SET_GPIO(GPIO_SPI0_CE0); // Disable Touch
+  SET_GPIO(GPIO_SPI0_CE1); // Disable Display
+  usleep(1000);
+
+  // Do the initialization with a very low SPI bus speed, so that it will succeed even if the bus speed chosen by the user is too high.
+  spi->clk = 34;
+  __sync_synchronize();
+
+  BEGIN_SPI_COMMUNICATION();
+  {
+    CLEAR_GPIO(GPIO_SPI0_CE0); // Enable Touch
+    CLEAR_GPIO(GPIO_SPI0_CE1); // Enable Display
+
+    BEGIN_SPI_COMMUNICATION();
+
+    usleep(25*1000);
+
+    SET_GPIO(GPIO_SPI0_CE0); // Disable Touch
+    usleep(25*1000);
+
+    SPI_TRANSFER(0x00001100); // Oy!  Now this is a reset
+    usleep(10*1000);
+    SPI_TRANSFER(0xff001100);
+    SPI_TRANSFER(0xff001100);
+    usleep(10*1000);
+    SPI_TRANSFER(0xff001100);
+    SPI_TRANSFER(0xff001100);
+    SPI_TRANSFER(0xff001100);
+    SPI_TRANSFER(0xff001100);
+    usleep(15*1000);
+    SPI_TRANSFER(0x11001100/*Sleep Out*/);
+    usleep(150*1000);
+
+    SPI_TRANSFER(0xB0001100, 0x00, 0x00); // CLK 30Hz?
+    SPI_TRANSFER(0xB3001100, 0x00, 0x02, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00); // set RGB?
+    SPI_TRANSFER(0xB9001100, 0x00, 0x01, 0x00, 0x00, 0x00, 0x0F, 0x00, 0x0f); // ext cmd?
+    SPI_TRANSFER(0xC0001100, 0x00, 0x13, 0x00, 0x3B, 0x00, 0x00, 0x00, 0x02
+	, 0x00, 0x00, 0x00, 0x01, 0x00, 0x00, 0x00, 0x43);                    // Timing?
+    SPI_TRANSFER(0xC1001100, 0x00, 0x08, 0x00, 0x0F, 0x00, 0x08, 0x00, 0x08); // gamma?
+    SPI_TRANSFER(0xC4001100, 0x00, 0x11, 0x00, 0x07, 0x00, 0x03, 0x00, 0x04); // sleep modes?
+    SPI_TRANSFER(0xC6001100, 0x00, 0x00); // ?
+    SPI_TRANSFER(0xC8001100, 0x00, 0x03, 0x00, 0x03, 0x00, 0x13, 0x00, 0x5C
+	, 0x00, 0x03, 0x00, 0x07, 0x00, 0x14, 0x00, 0x08
+	, 0x00, 0x00, 0x00, 0x21, 0x00, 0x08, 0x00, 0x14
+	, 0x00, 0x07, 0x00, 0x53, 0x00, 0x0C, 0x00, 0x13
+	, 0x00, 0x03, 0x00, 0x03, 0x00, 0x21, 0x00, 0x00); // ?
+    SPI_TRANSFER(0x35001100, 0x00, 0x00);
+
+#define MADCTL_BGR_PIXEL_ORDER (1<<3)
+#define MADCTL_ROW_COLUMN_EXCHANGE (1<<5)
+#define MADCTL_COLUMN_ADDRESS_ORDER_SWAP (1<<6)
+#define MADCTL_ROW_ADDRESS_ORDER_SWAP (1<<7)
+#define MADCTL_ROTATE_180_DEGREES (MADCTL_COLUMN_ADDRESS_ORDER_SWAP | MADCTL_ROW_ADDRESS_ORDER_SWAP)
+
+    uint8_t madctl = 0;
+#ifndef DISPLAY_SWAP_BGR
+    madctl |= MADCTL_BGR_PIXEL_ORDER;
+#endif
+#if defined(DISPLAY_FLIP_ORIENTATION_IN_HARDWARE)
+    madctl |= MADCTL_ROW_COLUMN_EXCHANGE;
+#endif
+#ifdef DISPLAY_ROTATE_180_DEGREES
+    madctl ^= MADCTL_ROTATE_180_DEGREES;
+#endif
+    //SPI_TRANSFER(0x36001100/*MADCTL: Memory Access Control*/, 0x00, madctl);
+    SPI_TRANSFER(0x36001100/*MADCTL: Memory Access Control*/, 0x00, 0x60);
```

Comment:
This looks like is dropping support for `madctl` configuration, which does display rotation and RGB-BGR mapping. Was there something wrong with the original line? It would be good to support the display rotations configuration.

Reply:
I couldn't get it to work.  I can retry.  It ought to work.

Reply:
Made an adjustment to one of the macros and it is fine.  Screen is opposite though -usb plug down- where it was usb plug up with 0x60 value (I couldn't get it to switch by controlling your macros but in the end, want plug down so, as it is.)